### PR TITLE
Bootstrap 3 Migration Prepwork

### PR DESCRIPTION
**Pull request by biyeun**
Will require `pip uninstall django-crispy-forms` (you can run `pip uninstall -r requirements/uninstall-requirements.txt` to acheive the same effect)

@dannyroberts @czue 
cc @proteusvacuum 
please start reviewing before this gets out of hand...I think this is good to get onto staging and do QA before I do any more work on this

What this is going to do:
- use a forked copy of crispy forms and inject a change into `thread_local.TEMPLATE_PACK` to determine whether or not to use the bootstrap 3 or bootstrap 2 template when loading a page.
- use the same `thread_local` method to update `thread_local.BOOTSTRAP_VERSION` to determine whether or not to bootstrap 3 or bootstrap 2 is being used during loading of page
- rename templates (add stems on frequently accessed core templates) to go under parallel bootstrap2 and bootstrap3 paths to alert devs modifying those templates that they need to update more copies
- minor cleanups

Related Submodule PRs:
- https://github.com/dimagi/django-crispy-forms/pull/2 [give a thumbs up when good, will replace master branch with this one]
- https://github.com/dimagi/commcarehq-prelogin/pull/16
- https://github.com/dimagi/ctable/pull/82

### CODE REVIEW

File: corehq/apps/settings/views.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from django.views.decorators.debug import sensitive_post_parameters
-from corehq.apps.style.decorators import check_preview_bootstrap3, use_select2
+from corehq.apps.style.decorators import use_bootstrap3, use_select2
```
Comment (dimagimon) [inline corehq/apps/settings/views.py]: F401 'use_bootstrap3' imported but unused

Comment (dimagimon) [inline corehq/apps/settings/views.py]: F401 'use_select2' imported but unused

Comment (snopoke) [inline corehq/apps/settings/views.py]: To do the uninstall you can add it to the
requirements/uninstall-requirements.txt file which gets processed during
deploy.

Comment (biyeun) [inline corehq/apps/settings/views.py]: yep, figured that one out when I deployed this branch to staging...thanks! :)

**Comment by czue** [comment]
thinking that @dannyroberts has more context than I do and that he should be primary reviewer but let me know if there's anything specific I should look at.

**Comment by dannyroberts** [comment]
@biyeun Build fail real:

```
======================================================================
ERROR: test_authenticate_as (corehq.apps.hqadmin.tests.test_authenticate_as.AuthenticateAsIntegrationTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/hqadmin/tests/test_authenticate_as.py", line 86, in test_authenticate_as
    resp = self.client.get(reverse(AuthenticateAs.urlname))
...
  File "submodules/bootstrap3_crispy/crispy_forms/templatetags/crispy_forms_tags.py", line 210, in render
    c = self.get_render(context)
  File "submodules/bootstrap3_crispy/crispy_forms/templatetags/crispy_forms_tags.py", line 127, in get_render
    actual_form.form_html = helper.render_layout(actual_form, node_context, template_pack=self.template_pack)
  File "submodules/bootstrap3_crispy/crispy_forms/helper.py", line 297, in render_layout
    template_pack=template_pack
  File "submodules/bootstrap3_crispy/crispy_forms/layout.py", line 134, in render
    template_pack=template_pack
  File "submodules/bootstrap3_crispy/crispy_forms/utils.py", line 79, in render_field
    return field.render(form, form_style, context, template_pack=template_pack)
  File "submodules/bootstrap3_crispy/crispy_forms/layout.py", line 198, in render
    return render_to_string(self.template, Context({'input': self}))
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/template/loader.py", line 162, in render_to_string
    t = get_template(template_name)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/template/loader.py", line 138, in get_template
    template, origin = find_template(template_name)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/template/loader.py", line 131, in find_template
    raise TemplateDoesNotExist(name)
TemplateDoesNotExist: bootstrap3/baseinput.html
```

**Comment by dannyroberts** [comment]
@czue yeah I think @biyeun and I will work together on getting this in. She walked me through the changes, and even though it touches a lot of files, it's largely just a few path renames and changing all the references. I've reviewed the changes in glue code more carefully as well.

**Comment by biyeun** [comment]
ok I think I fixed the build error (was due to a typo in crispy forms). waiting on that to finish.
I submitted a QA request, so hopefully that's set soon and then this will be good to merge.

**Comment by biyeun** [comment]
merge 
dimagi/commcarehq-prelogin#16
dimagi/ctable#82
and sign off on dimagi/django-crispy-forms#2
then i'll update submodules and then this can be merged in

QA is ALL SET
@dannyroberts

**Pull request merged**